app/controllers/documents_v2_controller.rb

```diff
@@ -104,7 +116,20 @@ def copy_shared
     end
   end
 
-  private
+  def create_access_keys(document)
+    # generate two unique new access keys - we can't use a unique index constraint because the keys will be null for older documents
+    # giving the length of the random strings this will probably never loop
+    read_access_key = nil
+    read_write_access_key = nil
+    loop do
+      read_access_key = SecureRandom.hex(20)
+      read_write_access_key = SecureRandom.hex(40)
+      break if !Document.find_by(read_access_key: read_access_key) && !Document.find_by(read_write_access_key: read_write_access_key)
+    end
+    document.read_access_key = read_access_key
+    document.read_write_access_key = read_write_access_key
+    document.run_key = read_write_access_key  # the run_key needs to be set because of the title validation: "validates :title, uniqueness: {scope: [:owner, :run_key]}"
```

Comment:
I think it would be better to disable this validation for owner less documents. So the validation would be changed to:
`validates :title, uniqueness: {scope: [:owner, :run_key]}, if: :owner?`

Then we don't have to set the run_key for new docs. And we also don't need to worry about unique titles in the new API.  Perhaps this means the `title=` param can be dropped from the API too?

If an author renames the master document and then updates the shared copy, how does the title get updated with this API?  Or does the CFM not actually care what the title is in the document store and instead it just pulls it from the document contents? If that is the case then perhaps we can drop the title field completely once the V1 API is gone.